src/Cake.Common/Tools/MSBuild/MSBuildFileLogger.cs

```diff
@@ -80,11 +82,12 @@ public MSBuildFileLogger()
         /// <summary>
         /// Process the file logger config and return parameters as a string.
         /// </summary>
+        /// <param name="environment">The environment.</param>
         /// <returns>The parameters separated by semi-colons.</returns>
-        public string GetParameters()
+        public string GetParameters(ICakeEnvironment environment)
         {
             var parameters = new List<string>();
-            parameters.Add(!string.IsNullOrWhiteSpace(LogFile) ? $"logfile={LogFile}" : null);
+            parameters.Add(LogFile != null ? $"logfile={LogFile.MakeAbsolute(environment).FullPath}" : null);
```

Comment:
I.e. This will fail:
```
C:\Users\mb\src\gh\cake>msbuild src/Cake.sln /t:Clean /fl /flp:logfile=C:\temp\A file name with whitespaces.txt;Verbosity=Diagnostic
Microsoft (R) Build Engine version 15.1.1012.6693
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1008: Only one project can be specified.
Switch: file

For switch syntax, type "MSBuild /help"
```

While this don't:
```
msbuild src/Cake.sln /t:Clean /fl /flp:logfile="C:\temp\A file name with whitespaces.txt";Verbosity=Diagnostic
```